azurerm/resource_arm_policy_definition.go

```diff
@@ -0,0 +1,218 @@
+				Optional:         true,
+				ValidateFunc:     validation.ValidateJsonString,
+				DiffSuppressFunc: structure.SuppressJsonDiff,
+			},
+
+			"meta_data": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				ValidateFunc:     validation.ValidateJsonString,
+				DiffSuppressFunc: structure.SuppressJsonDiff,
+			},
+
+			"parameters": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				ValidateFunc:     validation.ValidateJsonString,
+				DiffSuppressFunc: structure.SuppressJsonDiff,
+			},
+		},
+	}
+}
+
+func resourceArmPolicyDefinitionCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).policyDefinitionsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	name := d.Get("name").(string)
+	policyType := d.Get("policy_type").(string)
+	mode := d.Get("mode").(string)
+	displayName := d.Get("display_name").(string)
+	description := d.Get("description").(string)
+
+	properties := policy.DefinitionProperties{
+		PolicyType:  policy.Type(policyType),
+		Mode:        policy.Mode(mode),
+		DisplayName: utils.String(displayName),
+		Description: utils.String(description),
+	}
+
+	if policyRuleString := d.Get("policy_rule").(string); policyRuleString != "" {
+		policyRule, err := structure.ExpandJsonFromString(policyRuleString)
+		if err != nil {
+			return fmt.Errorf("unable to parse policy_rule: %s", err)
+		}
+		properties.PolicyRule = &policyRule
+	}
+
+	if metaDataString := d.Get("meta_data").(string); metaDataString != "" {
+		metaData, err := structure.ExpandJsonFromString(metaDataString)
+		if err != nil {
+			return fmt.Errorf("unable to parse meta_data: %s", err)
+		}
+		properties.Metadata = &metaData
+	}
+
+	if parametersString := d.Get("parameters").(string); parametersString != "" {
+		parameters, err := structure.ExpandJsonFromString(parametersString)
+		if err != nil {
+			return fmt.Errorf("unable to parse parameters: %s", err)
+		}
+		properties.Parameters = &parameters
+	}
+
+	definition := policy.Definition{
+		Name:                 utils.String(name),
+		DefinitionProperties: &properties,
+	}
+
+	_, err := client.CreateOrUpdate(ctx, name, definition)
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, name)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(*resp.ID)
+
+	return resourceArmPolicyDefinitionRead(d, meta)
+}
+
+func resourceArmPolicyDefinitionRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).policyDefinitionsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	name, err := parsePolicyDefinitionNameFromId(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] Error reading Policy Definition %q - removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("Error reading Policy Definition %+v", err)
+	}
+
+	d.Set("id", resp.ID)
```

Comment:
we don't need to set `id` here - since it's already set in the Create method above